### ISSUE TITLE
Use it for the `zigzag` example and document it in the `README#usage`

### ISSUE THREAD

**Issue by schomatis** [dignifiedquire/rust-gperftools#2]
Assignee: @schomatis.

Related to https://github.com/filecoin-project/rust-proofs/issues/481.

**Comment by schomatis** [comment]
The profile already exists in `zigzag`, but needs to be explicitly enabled:

```
cargo run --example zigzag --features profile --release -- --size 1 --no-bench
```

**Comment by dignifiedquire** [comment]
The feature is specific to rust-proofs, not to this tool

**Comment by schomatis** [comment]
(Also, and this might be relevant only to the Ubuntu 16 distribution I'm using, need to explicitly point to `/usr/local/lib/` to find the `libprofiler.so.0`: 
```
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib
```
)

**Comment by schomatis** [comment]
> The feature is specific to rust-proofs, not to this tool

Ok.

**Comment by dignifiedquire** [comment]
can you see if https://github.com/dignifiedquire/rust-gperftools/pull/1 fixes that for you?

**Comment by schomatis** [comment]
The CPU profiler works, trying to enable the heap profiler now the compiler fails to find the `heap_profiler` import although the IDE seems able to find it:

```
error[E0432]: unresolved import `gperftools::heap_profiler`
  --> filecoin-proofs/examples/zigzag.rs:19:17
   |
19 | use gperftools::heap_profiler::HEAP_PROFILER;
   |                 ^^^^^^^^^^^^^ could not find `heap_profiler` in `gperftools`
```

Looking into it.

**Comment by schomatis** [comment]
> can you see if #1 fixes that for you?

Sure, I'll give it a try.

**Comment by schomatis** [comment]
Can't seem to pass the feature, through the command line,

```
CARGO_INCREMENTAL=1 cargo build --release --example zigzag --features profile,heap
error: Package `proofs v0.1.0 (/home/user/go/src/github.com/filecoin-project/rust-proofs)` does not have these features: `heap`
```

`--all-features` also doesn't do the trick, I need to read more carefully the docs to see how to pass features to dependencies.

Need to specify the dependency of the feature: `--features profile,gperftools/heap`.

**Comment by schomatis** [comment]
> can you see if #1 fixes that for you?

~No, it doesn't :disappointed:~

~Ok, what's happening here (and this, again, might just be a problem of my Ubuntu environment) is that the PR is *working*, in the sense that `gperftools` itself is being built with `-l profiler -l tcmalloc` but not the zigzag `binary` (which I suppose is instrumented with the alternative `tcmalloc` library), that is, I can successfully build `gperftools` with or without `-l profiler` (with or without the PR) but that doesn't apply to the crate using it as a dependency.~

Scratch all that, the `/usr/local/lib/` not being in the search path is an Ubuntu (and Debian and similar) issue.

**Comment by schomatis** [comment]
I can enable the heap profile, as the output during the ZigZag test shows

```
Starting tracking the heap
```

but all the profile files end up empty.

For reproducibility purposes, this is the command I'm running:

```
rust-proofs/filecoin-proofs$ CARGO_INCREMENTAL=1 cargo build --release --example zigzag --features profile,gperftools/heap && ../target/release/examples/zigzag --size 1024 --no-bench
```

(should I be using bigger sizes? is there a way to reduce the sampling frequency? -if such a thing applies to the heap profiler-, the strange thing is that the files are empty, not even the header is found, in contrast when, for example, running a very short execution with the cpu profiler that later reports zero samples, but can open the profile file)

and this is the modified `zigzag` example to use the heap profiler: https://github.com/filecoin-project/rust-proofs/pull/487/commits/00048890a4d1a5202ef0a15df54679a7d69051f7

**Comment by dignifiedquire** [comment]
I have published v0.2 and added more documentation on how the heapprofiling can be controlled, especially in https://github.com/dignifiedquire/rust-gperftools/blob/master/src/heap_profiler.rs#L24 let me know if that helps

**Comment by schomatis** [comment]
Thanks, modifying the `HEAP_PROFILE_ALLOCATION_INTERVAL` env.var did the trick.
